crates/swc/tests/tsc-references/1.0lib-noErrors.1.normal.js

```diff
@@ -10,3 +10,4 @@
 //!  31 | /**
 //!  32 |   * Converts A string to an integer.
 //!     `----
+//!   x Syntax Error
```

Comment:
We should not have this, because we throw a `SyntaxError` from JS

Reply:
Emm, i think that we should have this.
Before this, `try_with_handler` get a function: `op: FnOnce(&Handler) -> Result<Ret, anyhow::Error>`, then return a `Result<Ret, anyhow:Error>`.
it would like this code
```rust
pub fn try_with_handler(...) {

   let ret = op(&handler);
   match ret {
     ...
      Err(err) => {
          return err.context(diagnostic_msg);
      }
   }   
}
```
In `tsc_references` tests, it's would print it without anyhow:error inner only print err's context by using api [err.to_string()](https://github.com/swc-project/swc/blob/main/crates/swc/tests/tsc.rs#L460-L462), So it would not print `Syntax Error`. Actually, it's should be printed.

Reply:
We had printed `SyntaxError` using context at the past, and I removed it because of an exact requirement

Reply:
I’m a bit confused now. So, in some situations we should print the SyntaxError message, while in other situations we should not print it? Otherwise, why not just drop it

Reply:
No, we should not print it. I mean I intentionally removed it.

Reply:
Ok, I got it.

Reply:
So we completely ignore the error return by op function? I fell a bit weird  :confused:.
```rust
    let handler = Handler::with_emitter(true, false, emitter);

    let ret = HANDLER.set(&handler, || op(&handler));

    match ret {
        Ok(ret) => {
            if diagnostics.contains_error() {
                return Err(diagnostics.as_wrapper(cm.clone(), config));
            }

            Ok(ret)
        }
        // ignore the error, only return diagnostics collected from handler.
        Err(_err) => Err(diagnostics.as_wrapper(cm.clone(), config)),
    }
```

Reply:
I don’t think we should, but we need a way to remove ‘SyntaxError’ from the error message of JS ‘SyntaxError’

Reply:
Ok, i got it.

Reply:
I think we may ignore empty errors, or create a special error type (unit struct like `struct OnlyDiagnostics;` in `swc_common` or `swc_error_reporters`.

Reply:
I don't get what you means. maybe you can give us a pseudo-code

Reply:
You can define an error struct named `OnlyDiagnostics` in the `swc_error_reporters`, and you can then [downcast](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#method.downcast) the error object to `OnlyDiagnostics`.

In `swc_error_reporters` crate

```rust
struct OnlyDiagnostics;

impl Error for OnlyDiagnostics {

}

// implement more traits, like Display


```

From `swc` crate (IIRC)

```rust

// Instead of `bail!("Syntax Error");`
anyhow::bail!(swc_error_reporters::OnlyDiagnostics);

```

From `try_xxx`

```rust
if let Some(..) = err.downcast_ref::<OnlyDiagnostics>() {
 // Drop contexts, and emit only diagnostics
} else {
 // Use err.to_string() to include contexts
}
```

Reply:
> You can define an error struct named OnlyDiagnostics in the swc_error_reporters, and you can then [downcast](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#method.downcast) the error object to OnlyDiagnostics.

I will try to implement it later

Reply:
Okay, I'll review it later today or tomorrow.

I'll change the target branch to `dev` instead of `main`, and I'll rebase-merge the `dev` branch into the main branch once all tasks are done. In this way all PRs will be a separate commit in the main branch git history.

For context of using another branch, merging the PR that makes JS throw `SyntaxError` with `Syntax Error` in error message will block updating `swc_core` (in next.js) or publishing a new version, so I would want to avoid it.